src/Shared/Microsoft.DotNet.Scaffolding.Shared/T4Templating/TextTemplatingEngineHost.cs

```diff
@@ -0,0 +1,149 @@
+using System;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using Microsoft.Extensions.DependencyModel;
+using Microsoft.VisualStudio.TextTemplating;
+
+namespace Microsoft.DotNet.Scaffolding.Shared.T4Templating
+{
+    /// <summary>
+    /// Enables callers to obtain an object denoting the current session.
+    /// A session represents series of executions of text templates.
+    /// The session object can be used to pass information from the host into the code of the text template.
+    /// </summary>
+    public class TextTemplatingEngineHost : ITextTemplatingSessionHost, ITextTemplatingEngineHost, IServiceProvider
```

Comment:
>TextTemplatingEngineHost [](http://example.com/codeflow?start=17&length=24)

I assume you will add tests in next iteration?

Reply:
yes, with some actual templates.